src/app/icd/server/BUILD.gn

```diff
@@ -66,13 +73,19 @@ source_set("manager") {
 
   public_deps = [
     ":configuration-data",
-    ":monitoring-table",
     ":notifier",
     ":observer",
-    ":sender",
     "${chip_root}/src/app:subscription-manager",
     "${chip_root}/src/credentials:credentials",
+    "${chip_root}/src/messaging",
```

Comment:
why is messaging now needed?

Reply:
This is because we to `#include <messaging/ExchangeMgr.h>` in the `ICDManager.h` for the Init method API purpose. Previously it was provided by the `ICDCheckinSender.h` but now it's not present for configuration not supporting CheckIn protocol.
For my perspective ideally it would be to not include it at all, as it's not used by SIT, but Mathieu stated it would be good to keep the same Init API for both configurations, even if some arguments are not used.

Reply:
I see. Ok, let's keep one init even if not used. It is probably be less confusing like that and should not impact code size much